src/ucp/api/ucp.h

```diff
@@ -1293,6 +1293,31 @@ ucs_status_t ucp_mem_map(ucp_context_h context, ucp_mem_map_params_t *params,
 ucs_status_t ucp_mem_unmap(ucp_context_h context, ucp_mem_h memh);
 
 
+enum ucp_mem_advice_flags {
+    UCP_MADV_NORMAL     = 0,         /* No special treatment */
+    UCP_MADV_WILLNEED   = UCS_BIT(0) /* Expect access in the near future */ 
+};
+/**
+ * @ingroup UCP_MEM
+ * @brief give advice about use of memory
+ *
+ * Advises the UCP about how to handle memory range beginning at address and
+ * size of length bytes. This call does not influence the semantics of the 
+ * application, but may influence its performance. The UCP is free to ignore
+ * the advice
+ *
+ * @param [in]  context     Application @ref ucp_context_h "context" which was
+ *                          used to allocate/map the memory.
+ * @param [in]  memh        @ref ucp_mem_h "Handle" to memory region.
+ * @param [in]  params      Memory base address and length. flag field is used
+ *                          to pass memory use advice as defined in 
+ *                          @ref ucp_mem_advice_flags
+ *
+ * @return Error code as defined by @ref ucs_status_t
+ */
+ucs_status_t ucp_mem_advise(ucp_context_h context, ucp_mem_h memh,  
+                            ucp_mem_map_params_t *params);
```

Comment:
pls use a different structure for this function, and not reuse the one for mmap. Unix's madvise() takes a "command" code and not "flags".
 Also, it could be passed as const *